### PR TITLE
BB-2028: Add new user console

### PR DESCRIPTION
**Pull request by giovannicimolin**
This PR lays the groundwork for the new Ocim frontend. It implements `login`, `logout` and a few calls to the backend to showcase the design.

The focus of this PR is on the frontend design implementation, so reducers, actions and other components that handle data are not fully implemented.

**Testing instructions:**
1. Review the contents of the `auth` and `console` folders. Most of the implementation is contained there.
2. Run the new frontend with `npm run start` and go to http://localhost:3000/.
3. Switch between the registration and login pages using the links in the header menu. There are a few known issues with the header styling that are not in the scope of this task, please ignore them for now.
4. Try to access the console page without being authenticated: http://localhost:3000/console
5. Go to the login page (http://localhost:3000/login) and click on `Login`, you should see a error message.
6. Try logging in with incorrect credentials.
7. Login with an existing account.
8. Check that the header now shows the console menu, compare it to the mockups.
9. Check that the multiple page elements are looking like the mockups (with the exception of the redeployment toolbar, that is going to be fully implemented on a follow up task).
10. Open a text editor and toggle the `loading` state values from the following files:
```
src/console/components/ConsolePage/ConsolePage.tsx
src/console/components/InstanceSettings/InstanceSettings.tsx
```
You should see the spinner when the page is loading and the correctly rendered page.
11. Check that the changes match the mockups to a good extent.

**Reviewier:**
- [ ] @xitij2000 

**Notes:**
There's still quite a few things to be improved:
1. Header styling.
2. Refresh token workflow (now handled by the PrivateRoute, but should be moved to the API client middleware).

### CODE REVIEW

File: frontend/src/console/components/RedeploymentToolbar/RedeploymentToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            <span className="text">Status: Up to date</span>
+          </div>
+        </OverlayTrigger>
+        <Button
+          className="float-right loading"
+          disabled
+          variant="primary"
+          size="lg"
+        >
+          Deploy (10 updates)
```
Comment (xitij2000) [inline frontend/src/console/components/RedeploymentToolbar/RedeploymentToolbar.tsx]: Should this be a message in the messages file? Also is 10 a parameter?

File: frontend/src/global/api.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // import { redirectToLogin } from 'auth/actions';
 import { OCIM_API_BASE } from 'global/constants';
 import { V2Api as _V2Api, Configuration } from 'ocim-client';
+// import { redirectToLogin } from 'auth/actions';
```
Comment (xitij2000) [inline frontend/src/global/api.ts]: nit: remove

File: frontend/src/auth/components/LoginPage/LoginPage.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+@connect<{}, ActionProps, {}, Props, RootState>(
+  (state: RootState) => ({
+    ...state.loginState
+  }),
+  {
+    performLogin
+  }
+)
+export class LoginPage extends React.PureComponent<Props, State> {
+  constructor(props: Props, state: State) {
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/auth/components/LoginPage/LoginPage.tsx]: @xitij2000 It works. The state is passed from connect using the interface defined just above in the file.

File: frontend/src/console/components/ConsolePage/ConsolePage.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            <i className="fas fa-sync-alt fa-spin" />
+          </h1>
+        </div>
+      );
+    }
+    const instanceData = this.props.instances[this.props.selectedInstance];
+    const instanceLink =
+      `https://${instanceData.subdomain}${INTERNAL_DOMAIN_NAME}` || '';
+    const studioLink =
+      `https://studio.${instanceData.subdomain}${INTERNAL_DOMAIN_NAME}` || '';
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/console/components/ConsolePage/ConsolePage.tsx]: I expected the backend to return the full urls and more information on the instance when reading from the apis.

Anyway, this task is just about the design/frontend implementation, improvements to the API will be done on a follow-up task (that's already in this sprint).

File: frontend/src/console/components/RedeploymentToolbar/RedeploymentToolbar.tsx (hunk 1 above)
Comment (giovannicimolin) [reply frontend/src/console/components/RedeploymentToolbar/RedeploymentToolbar.tsx]: The redeployment toolbar is just a placeholder and will be built from scratch on BB-2083.

File: frontend/src/auth/components/LoginPage/displayMessages.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    description: 'Forgot password link.'
+  },
+  username: {
+    defaultMessage: 'Username',
+    description: 'Username'
+  },
+  usernameHelp: {
+    defaultMessage: ' ',
+    description: 'Keep empty'
+  },
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/auth/components/LoginPage/displayMessages.ts]: I've reused the `TextInputField` component and it requires the `propnameHelp` key in the messages directory.
I've modified this component to fix the bug and will remove the empty message.

File: frontend/src/console/components/CustomizationSideMenu/CustomizationSideMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+  const customNavLink = (name: string, link: string) => (
+    <Nav.Link
+      active={currentLocation === link}
+      disabled={link === ''}
+      onClick={() => history.push(link)}
+    >
+      {name}
+    </Nav.Link>
+  );
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/console/components/CustomizationSideMenu/CustomizationSideMenu.tsx]: Done.

File: frontend/src/ui/components/App/styles.scss [hunk 7]
```diff
@@ -6,6 +6,6 @@
     width: 100%;
     background-image: url("~assets/header-background.jpg");
     background-color: $primary-1;
-    background-size: contain;
+    background-size: cover;
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/ui/components/App/styles.scss]: We need a better image for this (a large one). 
To fix the resizing issue, I've set `background-size: 100% 350px;`. It looks good on 4k and fullhd, but a little bit distorted in mobile. Should work for now, but we need to replace it in the future.

**Pull request merged**